proxy-node-gateway/src/main/java/uk/gov/ida/notification/session/GatewaySessionData.java

```diff
@@ -30,6 +38,26 @@ public GatewaySessionData(
         this.eidasDestination = eidasSamlParserResponse.getDestination();
         this.eidasRelayState = eidasRelayState;
         this.eidasConnectorPublicKey = eidasSamlParserResponse.getConnectorEncryptionPublicCertificate();
+        validate();
```

Comment:
When do the `@NotEmpty` validations get run? Feels like this stuff would be better lumped in all together so it checks `@NotEmpty` and runs your custom `validate` together. Don't know how easy that is given the framework is doing all the annotation based validation. Don't get me wrong this is functional, but it would be nice to have a common pattern for validating constructed objects.

Reply:
AFAIK, the annotation validations only get run when you call `validate()` on the object and nowhere else. I think that dropwizard will use them when serializing an object as the return of a resource, but outside of that you have to validate yourself.